datalad/support/tests/test_sshconnector.py

```diff
@@ -0,0 +1,61 @@
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Test classes SSHConnection and SSHManager
+
+"""
+
+import os
+from os.path import exists
+
+from nose.tools import ok_, assert_is_instance
+
+from datalad.support.sshconnector import SSHConnection, SSHManager
+from datalad.tests.utils import assert_raises
+from datalad.tests.utils import skip_if, skip_if_on_windows
+
+
+@skip_if(cond=not os.environ.get('DATALAD_TESTS_SSH'),
+         msg="Run this test by setting the DATALAD_TESTS_SSH")
+@skip_if_on_windows
+def test_ssh_get_connection():
+
+    manager = SSHManager()
+    c1 = manager.get_connection('ssh://localhost')
+    assert_is_instance(c1, SSHConnection)
+
+    # subsequent call returns the very same instance:
+    ok_(manager.get_connection('ssh://localhost') is c1)
+
+    # fail on malformed URls (meaning: urlparse can't correctly deal with them):
```

Comment:
Currently it should be possible, but would just need a leading `ssh://`.

I agree in general. Just need to find a proper way to parse whatever is passed as a url.